muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/users/index.ts

```diff
@@ -280,12 +301,22 @@ let loadStaff: LoadUsersTriggerType = function loadStaff(q?: string) {
   }
 }
 
-let loadUsers: LoadUsersTriggerType = function loadUsers(q?: string) {
+let loadUsers: LoadUsersTriggerType = function loadUsers(q: string | null, first: number | null, last: number | null) {
 
-  let query = { q: q };
+  let data = {
+    q: !q ? null : q,
+  };
 
-  let getStudents = q ? promisify(mApi().organizationUserManagement.students.read(query), 'callback')() : promisify(mApi().organizationUserManagement.students.read(), 'callback')();
-  let getStaffmembers = q ? promisify(mApi().organizationUserManagement.staffMembers.read(query), 'callback')() : promisify(mApi().organizationUserManagement.staffMembers.read(), 'callback')();
+  if (first) {
+    Object.assign(data, { firstResult: first });
```

Comment:
data.firstResult = first;

make data a type that allows for first result; that contains the shape that is supposed to hold.

eg.

data: {
  firstResult?: string;
  lastResult?: string;
  q: string;
}